tests/garage/tf/models/test_model.py

```diff
@@ -0,0 +1,216 @@
+import pickle
+
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.core.mlp import mlp
+from garage.tf.models.base import Model
+from tests.fixtures import TfGraphTestCase
+
+
+class SimpleModel(Model):
+    def __init__(self, output_dim=2, hidden_sizes=(4, 4), name=None):
+        super().__init__(name)
+        self._output_dim = output_dim
+        self._hidden_sizes = hidden_sizes
+
+    def network_output_spec(self):
+        return ['state', 'action']
+
+    def _build(self, obs_input):
+        state = mlp(obs_input, self._output_dim, self._hidden_sizes, 'state')
+        action = mlp(obs_input, self._output_dim, self._hidden_sizes, 'action')
+        return state, action
+
+
+class SimpleModel2(Model):
+    def __init__(self, output_dim=2, hidden_sizes=(4, 4), name=None):
+        super().__init__(name)
+        self._output_dim = output_dim
+        self._hidden_sizes = hidden_sizes
+
+    def _build(self, obs_input):
+        action = mlp(obs_input, self._output_dim, self._hidden_sizes, 'state')
+        return action
+
+
+class ComplicatedModel(Model):
```

Comment:
what happens if one of the constructor params is a model?

Reply:
Why mark this resolved? I’m also curious to know if it is possible to pass a simple model to a complicated model.

Reply:
sorry i resolved it and then asked the same question in a more relevant line below.

please resolve this yourself when you're satisfied :)

Reply:
no worry I will add a test for this